app/main.hs

```diff
@@ -104,21 +120,19 @@ main = do
     packages <- parseCabalFiles packageDirs ignoreList optsPackages
 
     dependencyByModule <- getDependencyByModule optsProjectRoot buildSystem packages
-    flip execStateT ExitSuccess $ for_ packages $ \T.Package {..} -> do
+    flip execStateT ExitSuccess $ for_ packages $ \package@T.Package {..} -> do
       let addSelf = if optsNoIgnoreSelf then id else Set.insert (T.DependencyName packageName)
       baseUsedDependencies <- fmap mconcat . for packageCompilables $ \compilable@T.Compilable {..} -> do
         usedDependencies <- addSelf <$> getCompilableUsedDependencies dependencyByModule compilable
         let (baseUsedDependencies, otherUsedDependencies) = Set.partition (flip Set.member packageBaseDependencies) usedDependencies
             otherUnusedDependencies = Set.difference compilableDependencies otherUsedDependencies
         unless (Set.null otherUnusedDependencies) $ do
-          liftIO . putStrLn . unpack $ "Some unused dependencies for " <> pack (show compilableType) <> " " <> T.unCompilableName compilableName <> " in package " <> packageName
-          traverse_ (liftIO . putStrLn . unpack . ("  " <>) . T.unDependencyName) $ Set.toList otherUnusedDependencies
-          put $ ExitFailure 1
+          shouldFail <- liftIO $ Unused.apply package otherUnusedDependencies (Just compilable) apply
```

Comment:
This runs for each target so it's gonna overwrite... I think we need `State` or something here.